x-pack/plugins/lens/public/datatable_visualization/visualization.tsx

```diff
@@ -232,4 +235,48 @@ export const datatableVisualization: Visualization<DatatableVisualizationState>
       ],
     };
   },
+
+  getErrorMessages(state, frame) {
+    const errors: Array<{
+      shortMessage: string;
+      longMessage: string;
+    }> = [];
+    if (state) {
+      const { sortedColumns, datasource } =
+        getDataSourceAndSortedColumns(state, frame.datasourceLayers, state.layers[0].layerId) || {};
+
+      // Data validation below
+      if (
```

Comment:
This kicks in on intial state with an empty datatable:
* Go to Lens
* Switch to table
* "Invalid configuration" is shown

I'm on the fence whether this is a "building state" (showing the "please drop field illustration") or an error state (showing the error message). If the user drags fields to the dimensions instead of the workspace, they will see this even if they didn't do anything wrong.

Your call if we want to treat it as a config error.
If yes: Please add a condition to this so it's not showing right away: `sortedColumns.length > 0`
If no: Let's remove this and simply return `null` instead of an expression in `toExpression` so no chart is rendered